sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -819,19 +819,32 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("cast string to boolean") {
-    checkCast("t", true)
+
     checkCast("true", true)
+    checkCast("tru", true)
+    checkCast("tr", true)
+    checkCast("t", true)
     checkCast("tRUe", true)
-    checkCast("y", true)
+    checkCast("    tRue   ", true)
+    checkCast("    tRu   ", true)
     checkCast("yes", true)
+    checkCast("ye", true)
+    checkCast("y", true)
     checkCast("1", true)
+    checkCast("on", true)
 
-    checkCast("f", false)
     checkCast("false", false)
-    checkCast("FAlsE", false)
-    checkCast("n", false)
+    checkCast("fals", false)
+    checkCast("fal", false)
+    checkCast("fa", false)
+    checkCast("f", false)
+    checkCast("    fAlse    ", false)
+    checkCast("    fAls    ", false)
     checkCast("no", false)
+    checkCast("n", false)
     checkCast("0", false)
+    checkCast("off", false)
+    checkCast("of", false)
```

Comment:
@younggyuchun, just for doubly sure, did you double check the behaviours against PostgreSQL?

Reply:
@HyukjinKwon Here it is:
![image](https://user-images.githubusercontent.com/1910477/63601427-62cd5900-c593-11e9-93bc-9a93c8a187d3.png)

Reply:
This is not documented: https://www.postgresql.org/docs/devel/datatype-boolean.html

Postgres may support `of` for history reasons, I don't think we have to follow it.

Reply:
@cloud-fan . It's a documented feature in that document. We had better support it.
> Unique prefixes of these strings are also accepted, for example t or n. Leading or trailing whitespace is ignored, and case does not matter.

cc @gatorsmile

Reply:
BTW, @younggyuchun . Please add a negative test case.
`o` is not supported because it's not unique. It's a common prefix for `on` and `off`.

Reply:
Seems okay to me

Reply:
SGTM